plaso/parsers/wincc.py

```diff
@@ -0,0 +1,345 @@
+    """Parses a bodyfile file-like object.
+
+    Args:
+      parser_mediator (ParserMediator): mediates interactions between parsers
+          and other components, such as storage and dfVFS.
+      file_object (dfvfs.FileIO): file-like object.
+
+    Raises:
+      WrongParser: when the file cannot be parsed.
+    """
+    line_reader = text_file.TextFile(
+        file_object, encoding=self.ENCODING, end_of_line=self.END_OF_LINE)
+
+    line_number = 0
+
+    try:
+      line = line_reader.readline()
+    except UnicodeDecodeError as exception:
+      raise errors.WrongParser(
+          'unable to read line: {0:d} with error: {1!s}'.format(
+              line_number, exception))
+
+    while line:
+      values = line.split(self.DELIMITER)
+      self._ParseValues(parser_mediator, line_number, values)
+
+      try:
+        line_number += 1
+        line = line_reader.readline()
+      except UnicodeDecodeError as exception:
+        parser_mediator.ProduceExtractionWarning(
+            'unable to read line: {0:d} with error: {1!s}'.format(
+                line_number, exception))
+        break
+
+
+class WinCCSysLogParser(interface.FileObjectParser):
+  """Text parser plugin for WinCC Sys Log files."""
+
+  NAME = 'wincc_sys'
+  DATA_FORMAT = 'WinCC Sys Log file'
+
+  DELIMITER = ','
+  ENCODING = 'utf-16-le'
+
+  def _ParseValues(self, parser_mediator, line_number, values, first_line):
+    """Parses WinCC log file values.
+
+    Args:
+      parser_mediator (ParserMediator): mediates interactions between parsers
+          and other components, such as storage and dfVFS.
+      line_number (int): number of the line the values were extracted from.
+      values (list[str]): values extracted from the line.
+      first_line (bool): True if this is first line from which values were
+          extracted.
+
+    Raises:
+      WrongParser: when the values cannot be parsed.
+    """
+
+    number_of_values = len(values)
+    if len(values) < 10:
+      error_string = 'invalid number of values : {0:d} in line: {1:d}'.format(
+          number_of_values, line_number)
+
+      # On other lines, we might encounter times where the split() operation
+      # produces more values.
+      if first_line:
+        raise errors.WrongParser(error_string)
+
+      parser_mediator.ProduceExtractionWarning(error_string)
+
+    event_data = WinCCSysLogEventData()
+
+    try:
+      log_identifier = int(values[0])
+      event_data.log_identifier = log_identifier
+    except ValueError as exception:
+      error_string = (
+          'Type of first value ({0!s}) should be an int in line: {1:d}').format(
+              values[0], line_number)
+      self._ParseValuesFail(
+          parser_mediator, first_line, exception, error_string)
+
+    try:
+      date_string = values[1]
+      time_string = values[2]
+      day_of_month, month, year = [
+          int(element) for element in date_string.split('.')]
```

Comment:
add the integer base e.g. `int(element, 10)`

Reply:
done (4 times)